<a href="https://colab.research.google.com/github/TVHM/summer_microcredentials_BUS/blob/main/mcPhearson__Unit3_Python_Advanced_Pandas_DataWrangling_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1 : Python Data Analysis using salaries dataset

# Data Exploration:  This should include summary statistics, means, medians, quartiles, or any other relevant information about the data set. 

In [31]:
import pandas as pd 
import numpy as np
import seaborn as sns

import matplotlib.pyplot as plt
%matplotlib inline



In [32]:
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/master/Salaries.csv')

In [33]:
#load data into a dataframe object

type(df)


pandas.core.frame.DataFrame

In [34]:
# 397 rows = 397 employees
df.shape

(397, 7)

In [35]:
df.head()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
0           1      Prof          B             19           18  Male  139750
1           2      Prof          B             20           16  Male  173200
2           3  AsstProf          B              4            3  Male   79750
3           4      Prof          B             45           39  Male  115000
4           5      Prof          B             40           41  Male  141500

In [36]:
df.tail()

Unnamed: 0      rank discipline  yrs.since.phd  yrs.service   sex  salary
392         393      Prof          A             33           30  Male  103106
393         394      Prof          A             31           19  Male  150564
394         395      Prof          A             42           25  Male  101738
395         396      Prof          A             25           15  Male   95329
396         397  AsstProf          A              8            4  Male   81035

In [37]:
df.describe()

Unnamed: 0  yrs.since.phd  yrs.service         salary
count  397.000000     397.000000   397.000000     397.000000
mean   199.000000      22.314861    17.614610  113706.458438
std    114.748275      12.887003    13.006024   30289.038695
min      1.000000       1.000000     0.000000   57800.000000
25%    100.000000      12.000000     7.000000   91000.000000
50%    199.000000      21.000000    16.000000  107300.000000
75%    298.000000      32.000000    27.000000  134185.000000
max    397.000000      56.000000    60.000000  231545.000000

# Mean/median of all salaries

In [38]:
# Mean of all salaries
df['salary'].mean()

113706.45843828715

In [39]:
# Median of all salaries
df['salary'].median()

107300.0

# Median salary between gender and/or rank(title) Etc...


Making a new Data Frame that only has the columns 'sex' and 'salary'

In [40]:
prime_df = df[['sex','salary']]

prime_df.head()


sex  salary
0  Male  139750
1  Male  173200
2  Male   79750
3  Male  115000
4  Male  141500

Finding the Median salary for all males

In [41]:
prime_df[prime_df['sex'] =='Male']['salary'].median()


108043.0

Finding the Median salary for all females

In [ ]:
prime_df[prime_df['sex'] =='Female']['salary'].median()

df2 = df[['sex','salary']].sort_values(by='salary').median


Comparing the Mean salary of Males and Females

In [ ]:
mean_females = prime_df[prime_df['sex']=="Female"]['salary'].mean()

In [ ]:
mean_males = prime_df[prime_df['sex']=="Male"]['salary'].mean()

In [ ]:
# The difference between mean salaries of male and female
mean_females-mean_males

# Data Wrangling – perform some basic transformations on the data

People who have 20 years of experience (Total: 157), (Mean Salary: 122,520.66)

In [ ]:
sum(df['yrs.service']>=20)

In [ ]:
df[df['yrs.service']>=20]['salary'].mean()

Comparing the salaries of people who have 20+ years of experience vs less than 20 years of experience

In [ ]:
mean_salary_20_more = df[df['yrs.service']>= 20]['salary'].mean()

In [ ]:
mean_salary_20_less = df[df['yrs.service']>= 20]['salary'].mean()

In [ ]:
mean_salary_20_more - mean_salary_20_less

People who have salary of greater than or equal to 200,000

In [ ]:

sum(df['salary'] >=200000)

In [ ]:
sum(df['yrs.since.phd'] <2)

In [44]:
#slaray of professors with more than 20 yrs since PhD.
df[df['yrs.since.phd']>=20]['salary'].mean()

125949.6129032258

In [46]:
#number of professors with les than 2 years service 
df[df['yrs.since.phd']<=2]['salary'].mean()

82075.44444444444

The sum number of professors with a minimum of one yr since PhD is equal to 4 earn  approx. 82,075.  That's 17,925 less than 126,000  salary earned by professors with more than 20 yrs since PhD.

# Rename values, subset columns, etc

In [ ]:
df.rename(columns = {'Unnamed; 0': 'ID'})

In [ ]:
# my rename column 
df.rename(columns = {'Unnamed; 0': 'ID'})


# Create some simple visualizations showing some simple statistics

In [ ]:
# visualization plot sns
df.plot(kind = 'scatter', x = 'yrs.service', y = 'salary')
plt.xlabel('years of service')
plt.ylabel('salary')

In [ ]:
df.plot(kind = 'scatter', x = 'discipline', y = 'salary')
plt.xlabel('discipline')
plt.ylabel('salary')

Your observations from the visuals: 

# Draw some conclusions based on the data.  What did you discover?



Based on an observation of mean and standard deviation, the avearge professor with 22 yrs since PhD, will have to work  20+ years more to earn almost as much as the highest paid professor with 56 yrs since PhD.

Other observations to look for include:


The relationship between salary and discipline indicates the highest salary is teaching within discipline B, yet the number of professors with higher salaries is greater withing discipline A.  




